### DISCUSSION TITLE
Suggest adding override String.IsInterned(Span<Char> span)

### DISCUSSION THREAD

**Discussion by Acerby** [dotnet/csharplang#6184] [Language Ideas] [answered]
In many scenarios when parsing strings we want to break them down using tokens which for which already exist as known constant string literals in within source code.  

Its fairly common to want to do Dictionary<string,Type> type lookups on Span<Char> values for performance and other reasons. However in many cases all the strings being looked for will already exist as interned string literals. Adding support for the above overload would potentially provide  a quick/simple way to avoid allocations in such scenarios, and would presumably be relatively simple to introduce as compared to adding full generalized support for enabling ref struct type lookups on dictionaries etc.

**Comment by PathogenDavid** [comment, marked as answer]
This is something for [the `runtime` repo](https://github.com/dotnet/runtime). It's been proposed before and was concluded to not be helpful: https://github.com/dotnet/runtime/issues/54546

If you disagree with this conclusion you should [open a new API proposal](https://github.com/dotnet/runtime/issues/new?assignees=&labels=api-suggestion&template=02_api_proposal.yml&title=%5BAPI+Proposal%5D%3A+).
